docs/infra/api-markdown-documenter/render-api-documentation.mjs

```diff
@@ -18,53 +18,59 @@ import chalk from "chalk";
 import fs from "fs-extra";
 import path from "path";
 
-import { alertNodeType } from "./alert-node.js";
-import { layoutContent } from "./api-documentation-layout.js";
-import { buildNavBar } from "./build-api-nav.js";
-import { renderAlertNode, renderBlockQuoteNode, renderTableNode } from "./custom-renderers.js";
-import { createHugoFrontMatter } from "./front-matter.js";
+import { admonitionNodeType } from "./admonition-node.mjs";
+import { layoutContent } from "./api-documentation-layout.mjs";
+import {
+	renderAdmonitionNode,
+	renderBlockQuoteNode,
+	renderTableNode,
+} from "./custom-renderers.mjs";
+
+const generatedContentNotice =
+	"<!-- Do not edit this file. It is automatically generated by @fluidtools/api-markdown-documenter. -->";
 
 /**
  * Generates a documentation suite for the API model saved under `inputDir`, saving the output to `outputDir`.
  * @param {string} inputDir - The directory path containing the API model to be processed.
  * @param {string} outputDir - The directory path under which the generated documentation suite will be saved.
  * @param {string} uriRootDir - The base for all links between API members.
- * @param {string} apiVersionNum - The API model version string used to differentiate different major versions of the
- * framework for which API documentation is presented on the website.
+ * @param {string} apiVersion - The "version" of the API model being processed, represented as a string.
+ * E.g. "1", "2", "2.1", etc.
+ * Used for some policy decisions, and for logging purposes.
  */
-export async function renderApiDocumentation(inputDir, outputDir, uriRootDir, apiVersionNum) {
+export async function renderApiDocumentation(inputDir, outputDir, uriRootDir, apiVersion) {
 	/**
 	 * Logs a progress message, prefaced with the API version number to help differentiate parallel logging output.
 	 */
 	function logProgress(message) {
-		console.log(`(${apiVersionNum}) ${message}`);
+		console.log(`(v${apiVersion}) ${message}`);
 	}
 
 	/**
 	 * Logs the error with the specified message, prefaced with the API version number to help differentiate parallel
 	 * logging output, and re-throws the error.
 	 */
 	function logErrorAndRethrow(message, error) {
-		console.error(chalk.red(`(${apiVersionNum}) ${message}:`));
+		console.error(chalk.red(`(v${apiVersion}) ${message}:`));
 		console.error(error);
 		throw error;
 	}
 
 	// Delete existing documentation output
 	logProgress("Removing existing generated API docs...");
 	await fs.ensureDir(outputDir);
-	await fs.emptyDir(outputDir);
+	// await fs.emptyDir(outputDir); // TODO: respect gitignore here
```

Comment:
Does this mean a render after removing some docs could end up keeping those files in the output?

Reply:
Oh, good catch. I need to fix this. Yes, it does have that effect, which isn't ideal.

Reply:
I filed a bug for the underlying issue: https://dev.azure.com/fluidframework/internal/_workitems/edit/24394

And introduced a temporary workaround in https://github.com/microsoft/FluidFramework/pull/22990/commits/84fe2f85c62375c6280cc9c21d7f32f1a7f5d5ed